### PR TITLE
README: sync required versions

### PR DESCRIPTION
**Pull request by kmilos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,15 +152,15 @@ Building
 ### Dependencies
 
 Compatible compilers/toolchains:
-* Clang: 12 and later
+* Clang: 14 and later
```
Comment (kmilos) [inline README.md:155]: This one depends on https://github.com/darktable-org/darktable/issues/15659

Comment (TurboGit) [reply README.md:155]: @kmilos : Indeed, depends on #15659. Given the message saying that Clang 12 could be the cause of an issue I'll certainly only allow for Clang 13.

Comment (kmilos) [reply README.md:155]: @TurboGit Which reminds me, can you please also fix https://github.com/darktable-org/darktable/commit/2fc3add89f9f7ce74509888af764d05b0128475d#r132715263

Comment (TurboGit) [reply README.md:155]: @kmilos : Done!

Comment (TurboGit) [reply README.md:155]: So please above change 14 to 13. TIA.

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**